src/wallet/wallet.cpp

```diff
@@ -2739,8 +2747,16 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CWalletT
                     return false;
                 }
 
-                LearnRelatedScripts(vchPubKey, g_change_type);
-                scriptChange = GetScriptForDestination(GetDestinationForKey(vchPubKey, g_change_type));
+                // If -changetype is not specified and g_address_type is not legacy,
+                // and any destination is (P2SH-)P2WPKH, use P2WPKH for the change
```

Comment:
shouldn't this read:
>// and any destination is P2WPKH, use P2WPKH for the change

Reply:
Actually I meant P2WPKH or P2WSH. Fixed.